### DISCUSSION TITLE
TestOps upload api not working.

### DISCUSSION THREAD

**Discussion by quldude** [allure-framework/allure2#1887] [TestOps]
I'm getting error when using below api in swagger page of TestOps. What am I doing wrong?

```
curl -X 'POST' \
  'https://xxxxx.testops.cloud/api/rs/launch/upload' \
  -H 'accept: */*' \
  -H 'Content-Type: multipart/form-data' \
  -H 'X-XSRF-TOKEN: {token}' \
  -F 'info={
  "name": "Launch3",
  "projectId": 1
}'

Response:
<!doctype html><html lang="en"><head><title>HTTP Status 415 – Unsupported Media Type</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 415 – Unsupported Media Type</h1></body></html>
```

Note: I tried with and without allure results files.

**Comment by cheshi-mantu** [comment]
hi @quldude 
could you please describe in more details what is it you are trying to achieve?

**Reply by quldude** [reply]
I'm trying to use the swagger api `launch/upload` to upload results into TestOps to create a launch. But swagger api response comes back as `HTTP Status 415 – Unsupported Media Type`.

**Comment by cheshi-mantu** [comment]
okay, I see. 
What is the global task which requires API usage?

please refer to our documentation on the API: https://docs.qameta.io/allure-testops/advanced/api/#limitations– it clearly states you can't use upload as it's a proprietary workflow.
Upload also does not create a launch. 
To upload the test results to Allure TestOps you either need to use a plug-in (if available, e.g. for [Jenkins](https://docs.qameta.io/allure-testops/integrations/jenkins/install-plugin/)) or [allurectl](https://docs.qameta.io/allure-testops/ecosystem/allurectl/).

**Reply by quldude** [reply]
I see allurectl has an upload option from here that https://docs.qameta.io/allure-testops/ecosystem/allurectl/ . Is there a way I can add environment variables, and links to a launch during the upload ie `allurectl upload ...` like how you have for tags `ALLURE_LAUNCH_TAGS`?

**Reply by cheshi-mantu** [reply]
Actually your choice should be `watch` not upload and yes, allurectl collects all env variables from the pipeline and sends this information to Allure TestOps launch, you just need to configure the processing of certain env variables on Allure TestOps side and you're all good.
See how it's done here: https://docs.qameta.io/allure-testops/briefly/launches/environment/
Works the same way for all the plugins and allurectl.

**Reply by cheshi-mantu** [reply]
here is an example: https://gitlab.com/cheshimantu/allure-pytest/-/blob/master/.gitlab-ci.yml#L9

**Reply by quldude** [reply]
what is the best way to add links for a launch when using allurectl?

**Reply by cheshi-mantu** [reply]
Launches don't have a link attribute, only name, tags, link to a issue tracker issue.

**Reply by cheshi-mantu** [reply]
if you need to add links to issue tracker issues, then please refer to this description of allurectl release: https://github.com/allure-framework/allurectl/releases/tag/2.2.1
